src/components/side-panel/side-panel.tsx

```diff
@@ -30,16 +38,18 @@ export const SidePanel = () => {
     };
 
     return (
-        <Wrapper $minimised={focusedElement ? false : !showSidePanel}>
-            <SidePanelHeader />
-            <ContentWrapper>
-                <ContentSection
-                    $isFocusedElement={focusedElement ? true : false}
-                >
-                    {renderPanelContent()}
-                </ContentSection>
-                {focusedElement === null && <Toolbar />}
-            </ContentWrapper>
-        </Wrapper>
+        <FormProvider {...methods}>
+            <Wrapper $minimised={focusedElement ? false : !showSidePanel}>
+                <SidePanelHeader />
+                <ContentWrapper>
+                    <ContentSection
+                        $isFocusedElement={focusedElement ? true : false}
+                    >
+                        {renderPanelContent()}
+                    </ContentSection>
+                    {focusedElement === null && <Toolbar />}
+                </ContentWrapper>
+            </Wrapper>
+        </FormProvider>
```

Comment:
- wrap the contents in `<form>` and have it handle the submit event
- this way, the submit button in `SidePanelHeader` just need to prevent html submission and does not need to handle the submit event

reason for handling here is because the bulk of the react-hook-form integration is here and the logic will be more visible here

```suggestion
        <FormProvider {...methods}>
            <form onSubmit={handleSubmit(onSubmit)()}>
                <Wrapper $minimised={focusedElement ? false : !showSidePanel}>
                    <SidePanelHeader />
                    <ContentWrapper>
                        <ContentSection
                            $isFocusedElement={focusedElement ? true : false}
                        >
                            {renderPanelContent()}
                        </ContentSection>
                        {focusedElement === null && <Toolbar />}
                    </ContentWrapper>
                </Wrapper>
            </form>
        </FormProvider>
```